### PR TITLE
[Reviewer: Ellie] Add health-checking infrastructure for exception handling

### PR DESCRIPTION
**Pull request by rkd-msw**
Ready for integration testing, but at the moment this regresses code coverage - I need to write a MockHealthChecker.

It's probably worth testing Homestead both with and without a HSS - I could potentially take on the HSS testing if that's more work than you were expecting.

### CODE REVIEW

**Comment by rkd-msw** [comment]
I've added UTs.

File: src/handlers.cpp [hunk 1]
```diff
@@ -534,6 +542,11 @@ void ImpiRegistrationStatusTask::run()
     writer.EndObject();
     _req.add_content(sb.GetString());
     send_http_reply(HTTP_OK);
+    printf("Value of _health_checker is %p\n", _health_checker);
```
Comment (eleanor-merry) [inline src/handlers.cpp]: Rogue printfs

File: src/ut/handlers_test.cpp [hunk 2]
```diff
@@ -2729,6 +2764,50 @@ TEST_F(HandlersTest, RegistrationStatus)
...
+  // Set up a request into the right URL.
+  MockHttpStack::Request req(_httpstack,
+                             "/impi/" + IMPI + "/",
+                             "registration-status",
+                             "?impu=" + IMPU);
+
+  ImpiRegistrationStatusTask::Config cfg(true);
+  ImpiRegistrationStatusTask* task = new ImpiRegistrationStatusTask(req, &cfg, FAKE_TRAIL_ID);
+
+  // Mock out a UAA response in order to triger a 200 OK return code.
```
Comment (eleanor-merry) [inline src/ut/handlers_test.cpp]: *trigger

**Comment by eleanor-merry** [comment]
Two picky comments. I'll pull it into sto1117 when it's ready.

**Comment by rkd-msw** [comment]
@eleanor-merry, it's ready - feel free to pull.

**Pull request closed without merging**